src/lib.rs

```diff
@@ -258,7 +260,19 @@ decl_error! {
         RemoteAttestationVerificationFailed,
         RemoteAttestationTooOld,
         ///The enclave cannot attest, because its building mode is not allowed
-        SgxModeNotAllowed
+        SgxModeNotAllowed,
+        ///The enclave is not registered
+        EnclaveIsNotRegistered,
+        ///The bonding account doesn't match the enclave
+        BondingAccountNotMatchEnclave,
+        ///The shard doesn't match the enclave
+        ShardNotMatchEnclave,
+        ///The worker url is too long
+        EnclaveUrlTooLong,
+        ///The RA report is too long
+        RaReportTooLong,
+        ///The enclave doesn't exists
+        EnclaveDoesntExist,
     }
```

Comment:
Here also try to remove the `to do` part. In encointer we use `InexistentEnclave`